sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -38,13 +38,16 @@ public ManagedIdentityClient(ManagedIdentityClientOptions options)
             }
 
             ClientId = options.ClientId;
+            ResourceIdentifier = options.ResourceIdentifier;
             Pipeline = options.Pipeline;
             _identitySource = new Lazy<ManagedIdentitySource>(() => SelectManagedIdentitySource(options));
         }
 
         internal CredentialPipeline Pipeline { get; }
 
-        protected string ClientId { get; }
+        internal protected string ClientId { get; }
+
+        internal ResourceIdentifier ResourceIdentifier { get; }
```

Comment:
Where is this used?

Reply:
It's used for validation in `DefaultAzureCredentialFactoryTests.ValidateManagedIdentityCtorOptionsHonored`.